crates/agentgateway/src/http/csrf.rs

```diff
@@ -0,0 +1,121 @@
+use ::http::{Method, StatusCode, header};
+use std::collections::HashSet;
+
+use crate::http::{PolicyResponse, Request, filters};
+use crate::*;
+
+#[apply(schema)]
+pub struct Csrf {
+	#[serde(default)]
+	additional_origins: HashSet<String>,
+}
+
+impl Csrf {
+	pub fn apply(&self, req: &mut Request) -> Result<PolicyResponse, filters::Error> {
+		// 1. Allow all GET, HEAD, or OPTIONS requests
+		if is_safe_method(req.method()) {
+			return Ok(Default::default());
+		}
+
+		// 2. Check Sec-Fetch-Site header first
+		match get_sec_fetch_site_header(req)? {
+			Some(sec_fetch_site) => {
+				match sec_fetch_site.as_str() {
+					"same-origin" | "none" => return Ok(Default::default()),
+					_ => {
+						// Check if request is exempt (trusted origins)
+						if self.is_request_exempt(req)? {
+							return Ok(Default::default());
+						}
+						return self.create_forbidden_response(
```

Comment:
Let's log the detailed reason internally for debugging (Sec-Fetch-Site, Origin mismatch, etc.) but don’t expose it to the client